src/utils/types.ts

```diff
@@ -1 +1,5 @@
-export default {};
+export class UnreachableCaseError extends Error {
+  constructor(value: never) {
+    super(`Unreachable case: ${value}`);
+  }
+}
```

Comment:
This should either be an `AssertionError` or a `MetaplexError` that should be moved to the `errors` folder. I don't want us to support yet another type of error.